### ISSUE TITLE
Kani compiler should give a warning when user code includes code that we cannot handle

### ISSUE THREAD

**Issue by celinval** [model-checking/kani#922]
Requested feature: Kani should be upfront with the user about any construct that it cannot reason about. Users should be aware that the verification may fail because we cannot compile certain constructs.
Use case: User code include things like inline assembly.
Link to relevant documentation (Rust reference, Nomicon, RFC):
Is this a breaking change?

Test case:

```rust
pub fn asm() {
    unsafe {
        std::arch::asm!("NOP");
    }
}

#[kani::proof]
fn harness_fail() {
    asm();
}

#[kani::proof]
fn harness_succeed() {
    assert!(vec![].len() == 0);
}
```

When compiling this crate, I would expect to see a warning saying that we cannot codegen` std::arch::asm!("NOP");`. Additionally, we should keep the behavior where the verification fails if the construct is reachable (`harness_fail`) and succeed if it's not reachable (`harness_succeed`).

**Comment by zhassan-aws** [comment]
No warnings/errors are emitted in codegen for:
```rust
use std::thread;

fn main() {
    let thread_join_handle = thread::spawn(move || {
        // some work here
    });
    // some work here
    let res = thread_join_handle.join();
}
```
